delivery/repositories.py

```diff
@@ -42,8 +44,43 @@ def update(self, content: Restaurant) -> None:
     def get_all(self) -> Iterable[Restaurant]:
         return self.restaurants
 
-    def get_by_id(self, restaurant_id: int) -> Restaurant:
+    def get_by_id(self, restaurant_id: str) -> Restaurant:
         for restaurant in self.restaurants:
-            if restaurant.id == restaurant_id:
+            if restaurant._id == restaurant_id:
                 return restaurant
         raise WrongIdError(description="Restaurant with such ID doesn't exist")
+
+
+class MongoRepository(AbstractRestaurantRepository):
+    def __init__(self, restaurants):
+        self.restaurants = restaurants
+
+    def create(self, content: Restaurant):
+        content = asdict(content)
+        content.pop("_id")
+        self.restaurants.restaurant.insert_one(content)
+
+    def update(self, content: Restaurant):
+        update_data = asdict(content)
+        try:
+            restaurant = {'_id': ObjectId(update_data["_id"])}
+        except:
+            raise WrongIdError(description="Restaurant with such ID doesn't exist")
+        for key, value in asdict(content).items():
+            if value == "" or key == "_id":
+                update_data.pop(key)
+        self.restaurants.restaurant.update_one(restaurant, {"$set": update_data})
+
+    def get_all(self):
+        restaurants_list = []
+        for restaurant in self.restaurants.restaurant.find({}):
+            restaurants_list.append(Restaurant(**restaurant))
+        return restaurants_list
+
+    def get_by_id(self, restaurant_id):
+        try:
+            restaurant = self.restaurants.restaurant.find_one({"_id": ObjectId(restaurant_id)})
```

Comment:
`find_one` will just return `None` if the object was not found, so `WrongIdError` is not handled properly here

Reply:
fixed